src/app/components/Document/index.jsx

```diff
@@ -5,14 +5,44 @@ import { C_POSTBOX } from '../../lib/constants/styles';
 
 /* eslint-disable react/prop-types */
 const Document = ({ assets, app, data, styleTags, helmet }) => {
+  const isAmpPayload = () => helmet.htmlAttributes.toString().includes('amp');
+
+  /* eslint-disable react/no-danger */
+  /* disabling the rule that bans the use of dangerouslySetInnerHTML until a more appropriate implementation can be implemented */
+  const injectClientSideApp = serialisedData => {
+    if (serialisedData) {
+      return (
+        <script
+          dangerouslySetInnerHTML={{
+            __html: `window.SIMORGH_DATA=${serialisedData}`,
+          }}
+        />
+      );
+    }
+
+    return null;
+  };
+
   const htmlAttrs = helmet.htmlAttributes.toComponent();
   const meta = helmet.meta.toComponent();
   const title = helmet.title.toComponent();
   const links = helmet.link.toComponent();
-  const serialisedData = JSON.stringify(data);
-  const scripts = assets.map(asset => (
-    <script key={asset} type="text/javascript" src={asset} defer />
-  ));
+  let scripts = [];
+  let ampScript;
+  let serialisedData;
+
+  if (isAmpPayload()) {
+    ampScript = (
+      <script key="amp" async src="https://cdn.ampproject.org/v0.js" />
+    );
+  }
+
+  if (!isAmpPayload()) {
```

Comment:
Keeping this as a seperate if statement, not an else, as we are likely to have additional requirements to conditionally inject the client JS or application.